### ISSUE TITLE
Unexpected ValidationError from pydantic for field with compound type

### ISSUE THREAD

**Issue by agrav** [koxudaxi/datamodel-code-generator#216]
**Describe the bug**
When instantiating a data model from generated code an unexpected ValidationError is raised from pydantic.

**To Reproduce**

Example schema:

```json

{
  "openapi": "3.0.2",
  "info": {
    "title": "A title",
    "description": "A description",
    "version": "0.9.0"
  },
  "components": {
    "schemas": {
      "FileSetUpload": {
        "title": "FileSetUpload",
        "required": [
          "tags"
        ],
        "type": "object",
        "properties": {
          "task_id": {
            "title": "Task id",
            "type": "string"
          },
          "tags": {
            "title": "Dict of tags, each containing a list of file names",
            "type": "object",
            "additionalProperties": {
              "type": "array",
              "items": {
                "type": "string"
              }
            }
          }
        }
      }
    },
    "securitySchemes": {
      "JWTBearer": {
        "type": "http",
        "scheme": "bearer"
      }
    }
  }
}

```

Used commandline:
```
$ datamodel-codegen --input openapi_test.json --output models_test.py
```

**Expected behavior**

From the schema the following code is generated:

```python
# generated by datamodel-codegen:
#   filename:  openapi_test.json
#   timestamp: 2020-09-15T05:45:08+00:00

from __future__ import annotations

from typing import Dict, Optional

from pydantic import BaseModel, Field


class Tags(BaseModel):
    pass


class FileSetUpload(BaseModel):
    task_id: Optional[str] = Field(None, title='Task id')
    tags: Dict[str, Tags] = Field(
        ..., title='Dict of tags, each containing a list of file names'
    )
```

When instantiating this class with the following code:

```python
FileSetUpload(tags={'tag1': ['filename1', 'filename2'], 'tag2': ['filename3', 'filename4']})
```

The following error is raised from pydantic:

```
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "pydantic\main.py", line 346, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 2 validation errors for FileSetUpload
tags -> tag1
  value is not a valid dict (type=type_error.dict)
tags -> tag2
  value is not a valid dict (type=type_error.dict)
```

Expected behaviour:

```python
from __future__ import annotations

from typing import Dict, Optional, List

from pydantic import BaseModel, Field


class FileSetUpload(BaseModel):
    task_id: Optional[str] = Field(None, title='Task id')
    tags: Dict[str, List[str]] = Field(
        ..., title='Dict of tags, each containing a list of file names'
    )
```

**Version:**
 - OS: Windows 10
 - Python Verison: 3.7.7
 - datamodel-code-generator Version 0.5.31

**Comment by koxudaxi** [comment]
Thank you for creating this issue.
You expect `Tag` is not model and the type is List[str].
But, I think we should treat `Tag` as a model.
How do you think about it?
I put an example.
 

```
from pydantic import BaseModel, Field


class Tags(BaseModel):
    __root__: List[str]


class FileSetUpload(BaseModel):
    task_id: Optional[str] = Field(None, title="Task id")
    tags: Dict[str, Tags] = Field(
        ..., title="Dict of tags, each containing a list of file names"
    )


model = FileSetUpload(
    tags={"tag1": ["filename1", "filename2"], "tag2": ["filename3", "filename4"]}
)

print(model)
# task_id=None tags={'tag1': Tags(__root__=['filename1', 'filename2']), 'tag2': Tags(__root__=['filename3', 'filename4'])}

print(model.dict())
# {'task_id': None, 'tags': {'tag1': ['filename1', 'filename2'], 'tag2': ['filename3', 'filename4']}}
```

**Comment by agrav** [comment]
Thank you for answering.
That works fine for me !
The `openapi.json` that I use is generated by FastAPI, which in turn uses the `List[str]` in the original pydantic model. So that was my only preference. I guess the `Tags` model is better for the codegen.

**Comment by koxudaxi** [comment]
@agrav 
I have fixed the problem. And I have released a new version as `0.5.32`.
